src/core/metrics/sources/nginx_plus.go

```diff
@@ -53,31 +59,39 @@ func NewNginxPlus(baseDimensions *metrics.CommonDim, nginxNamespace, plusNamespa
 func (c *NginxPlus) Collect(ctx context.Context, wg *sync.WaitGroup, m chan<- *metrics.StatsEntityWrapper) {
 	defer wg.Done()
 	c.init.Do(func() {
-		cl, err := plusclient.NewNginxClientWithVersion(&http.Client{}, c.plusAPI, c.clientVersion)
+		latestAPIVersion, err := getLatestAPIVersion(ctx, &http.Client{}, c.plusAPI)
+		if err != nil {
+			c.logger.Log(fmt.Sprintf("Failed to check available api versions: %v", err))
```

Comment:
if there is an error, can you try use the default client version? https://github.com/nginx/agent/blob/main/src/core/config/defaults.go#L64 this should be 7